src/Adapter/Category/CommandHandler/UpdateCategoriesStatusHandler.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Category\CommandHandler;
+
+use Category;
+use PrestaShop\PrestaShop\Core\Domain\Category\Command\UpdateCategoriesStatusCommand;
+use PrestaShop\PrestaShop\Core\Domain\Category\CommandHandler\UpdateCategoriesStatusHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CannotUpdateCategoryStatusException;
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryStatus;
+
+/**
+ * Class ChangeCategoriesStatusHandler.
+ *
+ * @internal
+ */
+final class UpdateCategoriesStatusHandler implements UpdateCategoriesStatusHandlerInterface
```

Comment:
This class raises an interesting question: should we have a UpdateCategoryStatusHandler (handles 1 category) and a UpdateCategoriesStatusHandler (handles multiple categories) ? Or do we always use the multiple UpdateCategoriesStatusHandler ? Then should we do that for all command handlers (always handle the multiple usecase) ?
@eternoendless @mickaelandrieu

Reply:
i've discussed similar topic (Delete & Bulk delete) with @eternoendless when doing commands for SqlRequest. We ended up using 2 different handlers `DeleteSqlRequestHandler` and `BulkDeleteSqlRequestHandler`